### ISSUE TITLE
Is the feature of copying generators deprecated?

### ISSUE THREAD

**Issue by PabloRuizCuevas** [pypy/pypy#5153]
Before, in the version 2.7 using the flag withmod-_pickle_support we could copy generators with the full state. This is a really cool feature of pypy that I was trying to push for cpython, as it allows to use generators in multiprocessing and even in another machine, I think originally inspired in stackless python as I see in the docs.

https://discuss.python.org/t/copy-a-generator-with-state/73859/11

Running:

`pypy ../../rpython/bin/rpython --opt=2 targetpypystandalone.py –withmod-_pickle_support`

with the code

```python
>>>> import platform
>>>> platform.python_implementation()
'PyPy'
>>>> import pickle
>>>> def g():
....     for i in xrange(10):
....         yield i
....
>>>> gen = g()
>>>> next(gen)
0
>>>> next(gen)
1
>>>> pickled = pickle.dumps(gen)
>>>> new_gen = pickle.loads(pickled)
>>>> next(new_gen)
2
>>>> next(gen)
2
>>>> next(new_gen)
3
>>>> next(new_gen)
4
>>>> next(gen)
3
>>>> another_new_gen = pickle.loads(pickled)
>>>> next(another_new_gen)
2
```

bifurcates the generator, without saving the intermediate values (like itertools.tee does), which is what it was intended, so very cool, but for newer versions seems to throw an error: 

`pickle.PicklingError: Can't pickle <class 'generator'>: it's not found as builtins.generator`

Is it intended?

**Comment by cfbolz** [comment]
Yeah, we unfortunately never ported the feature to pypy3. It would probably not be super hard to get it back, and I've always been sad about us never getting around to it. Otoh, the demand for the feature has been low so far, you are the first person to point it out.

Still, if it's useful for you as an argument towards cpython, I'm happy to try to see how annoying it is. Probably only next year, though.

Cheers,

CF

**Comment by PabloRuizCuevas** [comment]
Cool, that would be amazing.

It is a feature that is not very often needed but when is needed it can make a huge difference.

Regards.
